test/jdk/javax/swing/JComboBox/bug4530953.java

```diff
@@ -0,0 +1,94 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.FlowLayout;
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @bug 4530953
+ * @summary Tests that highlighted Item appears after automatically scrolling to the item
+ * @key headful
+ * @run main bug4530953
+ */
+
+public class bug4530953 {
+    static JFrame frame;
+    static JComboBox combo;
+    static String[] data = { "Apple", "Orange", "Cherry" };
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        SwingUtilities.invokeAndWait(() -> createTestUI());
+        robot.waitForIdle();
+        robot.delay(250);
+
+        // enter some text in combo box editor
+        robot.keyPress(KeyEvent.VK_A);
+        robot.keyRelease(KeyEvent.VK_A);
+        robot.keyPress(KeyEvent.VK_A);
+        robot.keyRelease(KeyEvent.VK_A);
+        robot.keyPress(KeyEvent.VK_ENTER);
+        robot.keyRelease(KeyEvent.VK_ENTER);
+        robot.waitForIdle();
+        robot.delay(250);
+
+        // select orange in combo box
+        robot.keyPress(KeyEvent.VK_DOWN);
+        robot.keyRelease(KeyEvent.VK_DOWN);
+        robot.keyPress(KeyEvent.VK_DOWN);
+        robot.keyRelease(KeyEvent.VK_DOWN);
+        robot.keyPress(KeyEvent.VK_DOWN);
+        robot.keyRelease(KeyEvent.VK_DOWN);
+        robot.keyPress(KeyEvent.VK_ENTER);
+        robot.keyRelease(KeyEvent.VK_ENTER);
+        robot.waitForIdle();
+        robot.delay(250);
+
+        String currSelection = (String) combo.getEditor().getItem();
+        if (!currSelection.equals("Orange")) {
+            throw new RuntimeException("Unexpected Selection.\n" +
+                    "Expected: Orange\nActual: " + currSelection);
+        }
+
+        if (frame != null) {
+            SwingUtilities.invokeAndWait(() -> frame.dispose());
+        }
+    }
+
+    public static void createTestUI() {
+        frame = new JFrame("bug4530953");
+        combo = new JComboBox(data);
+        combo.setEditable(true);
+        combo.setSelectedIndex(1);
+        frame.getContentPane().setLayout(new FlowLayout());
+        frame.getContentPane().add(combo);
+        frame.setLocationRelativeTo(null);
+        frame.pack();
```

Comment:
guess setLocationRelativeTo should be called after pack..there was an issue if it is called before..